spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -110,18 +111,30 @@ public GcpPubSubAutoConfiguration(GcpPubSubProperties gcpPubSubProperties,
 		}
 	}
 
+	@Bean
+	@ConditionalOnMissingBean(name = "pubsubPublisherThreadPool")
+	public ScheduledExecutorService pubsubPublisherThreadPool() {
+		return Executors.newScheduledThreadPool(this.gcpPubSubProperties.getSubscriber().getExecutorThreads());
```

Comment:
What is wrong with Spring's `ThreadPoolTaskScheduler` ?

Reply:
Nothing :)